Python/specialize.c

```diff
@@ -935,8 +935,7 @@ analyze_descriptor_load(PyTypeObject *type, PyObject *name, PyObject **descr, un
         PyObject *getattr = _PyType_Lookup(type, &_Py_ID(__getattr__));
         has_getattr = getattr != NULL;
         if (has_custom_getattribute) {
-            if (getattro_slot == _Py_slot_tp_getattro &&
-                !has_getattr &&
+            if (!has_getattr &&
```

Comment:
@Fidget-Spinner, I removed the `getattro_slot == _Py_slot_tp_getattro` check here because:

1) It would prevent specialization of `__getattribute__` Python calls in the free threaded build with this change
2) I don't think it's necessary. The combined `has_custom_getattribute` and `!has_getattr` checks are sufficient.

Does this make sense to you?

Reply:
It makes sense for the FT build. Just reasoning out loud here: From what I see above, we replace the NULL slot with the special "`__getattribute__` but no `__getattr__`" slot at some lookup time. But it's functionally the same as just saying `get_attr == NULL`. So the check for the slot is redundant?